test/behavior/switch.zig

```diff
@@ -652,3 +656,8 @@ test "switch capture copies its payload" {
     try S.doTheTest();
     comptime try S.doTheTest();
 }
+
+test "switch on empty error set" {
+    var a: error{} = undefined;
+    switch (a) {}
```

Comment:
Pretty sure this is Illegal Behavior because according to the proposed spec, even 0-bit types can theoretically hold the value `undefined` and branching on `undefined` is Illegal Behavior.

An empty error set is effectively the same thing as an empty enum, which is effectively the same thing as `noreturn`. Switching on an empty error set is equivalent to `unreachable`.

Alternate proposed behavior test:

```zig
test {
    if (foo()) |_| {
        // OK
    } else |err| switch (err) {}
}
fn foo() !void {}
```

Reply:
I did realize this after seeing the test failure (I didn't run the behavior tests on stage1 when developing) and realized this was actually the issue I was seeing. I tried to also apply the fix in #11126 for inferred error sets but ran into some issues doing so.

I'm guessing an empty switch should unconditionally be a compile error (if semantically analyzed)?